pkgs/development/compilers/carp/default.nix

```diff
@@ -3,13 +3,13 @@
 haskellPackages.mkDerivation rec {
 
   pname = "carp";
```

Comment:
> may I ask what formatter you use?

nixpkgs-fmt

> I always format at before submitting using nixfmt

Please use nixpkgs-fmt. nixfmt enforces very short lines and unnecessary line breaks for long strings like url.
Also both don't do proper whitespace fixing.

> IMO aren't beyond personal preferences.

That is not true. Having a uniform style makes it easier for new contributors to know how things are done and treewide changes are also easier done because you don't need to account for variances. 

Also from platforms the darwin can be removed. unix consists of linux and darwin.